app/frontend/stores/session-store.ts

```diff
@@ -0,0 +1,69 @@
+import { flow, Instance, toGenerator, types } from "mobx-state-tree"
+import { withEnvironment } from "../lib/with-environment"
+import { withRootStore } from "../lib/with-root-store"
+
+export const SessionStoreModel = types
+  .model("SessionStoreModel")
+  .props({
+    loggedIn: types.optional(types.boolean, false),
+  })
+  .extend(withEnvironment())
+  .extend(withRootStore())
+  .views((self) => ({}))
+  .actions((self) => ({
+    resetAuth: flow(function* () {
+      self.loggedIn = false
+    }),
+  }))
+  .actions((self) => ({
+    handleLogin(response) {
+      if (response.ok) {
+        const user = response.data.data
+        self.loggedIn = true
+        self.rootStore.userStore.setCurrentUser(user)
+        return true
+      }
+      return false
+    },
+    handleForgotPasswordRequest: flow(function* (params) {
+      const response = yield self.environment.api.requestPasswordReset(params)
+      if (response.ok) {
+        return true
+      }
+    }),
+  }))
+  .actions((self) => ({
+    validateToken: flow(function* () {
```

Comment:
we should call `validateToken` in a `useEffect` on the navigation component to check and see if there is an authenticated user (e.g. once logged in, if the page is reloaded the user should not have to re-login OR if the cookie is invalid, the user should be logged out and see the login page) - see PM for reference

Reply:
Done and done